m3/config.go

```diff
@@ -45,6 +45,9 @@ type Configuration struct {
 
 	// IncludeHost is whether or not to include host tag.
 	IncludeHost bool `yaml:"includeHost"`
+
+	// SanitizeMetrics is whether or not metric identifiers (name, tags) are sanitized.
+	SanitizeMetrics bool `yaml:"sanitizeMetrics"`
```

Comment:
I'm supportive of this as well, I think by default just having a `nil` value for `SanitizeFn` is best and we can provide a `SanitizeAlphanumericFn` and a `SanitizeStatsdFn` as out of the box sanitizers people can set the `SanitizeFn` in the Scope options too.